superset/security/manager.py

```diff
@@ -1097,6 +1105,12 @@ def get_user_by_username(
     def get_anonymous_user(self) -> User:  # pylint: disable=no-self-use
         return AnonymousUserMixin()
 
+    def get_user_roles(self) -> List[Role]:
+        if g.user.is_anonymous:
```

Comment:
Actually I guess maybe we didn't need to change this method after all. We were considering a version of `GuestUser` where `is_anonymous == True`, which would have required changing this. That guest user implementation ended up being too unwieldy, but this is still cleaner than before so I'm keeping it.